derp/derphttp/derphttp_client.go

```diff
@@ -88,11 +90,12 @@ func (c *Client) String() string {
 
 // NewRegionClient returns a new DERP-over-HTTP client. It connects lazily.
 // To trigger a connection, use Connect.
-func NewRegionClient(privateKey key.NodePrivate, logf logger.Logf, getRegion func() *tailcfg.DERPRegion) *Client {
+func NewRegionClient(privateKey key.NodePrivate, logf logger.Logf, mon *netmon.Monitor, getRegion func() *tailcfg.DERPRegion) *Client {
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
document that this can be nil (apparently) and what nil means

Maybe:

    // The mon parameter is optional; if non-nil it's used to do faster interface lookups.